gulp-tasks/build-packages.js

```diff
@@ -1,204 +1,59 @@
-  const outputDirectory = path.join(packagePath,
-    constants.PACKAGE_BUILD_DIRNAME, constants.BROWSER_BUILD_DIRNAME);
-
-  logHelper.log(oneLine`
-    Building Browser Bundle for
-    ${logHelper.highlight(packageName)}.
-  `);
-  logHelper.log(`    Namespace: ${logHelper.highlight(namespace)}`);
-  logHelper.log(`    Filename: ${logHelper.highlight(outputFilename)}`);
-
-  return rollupStream({
-    input: moduleIndexPath,
-    rollup,
-    format: 'iife',
-    exports: 'named',
-    name: namespace,
-    sourcemap: true,
-    globals,
-    external: externalAndPure,
-    pureExternalModules: externalAndPure,
-    plugins: rollupHelper.getDefaultPlugins(buildType),
-    onwarn: (warning) => {
-      if (buildType === 'production' &&
-        warning.code === 'UNUSED_EXTERNAL_IMPORT') {
-        // This can occur when using rollup-plugin-replace.
-        logHelper.warn(`[${warning.code}] ${warning.message}`);
-        return;
-      }
-
-      // The final builds should have no warnings.
-      throw new Error(`Unhandled Rollup Warning: [${warning.code}] ` +
-        `'${warning.message}'`);
-    },
-  })
-  .on('error', (err) => {
-    const args = [];
-    Object.keys(err).forEach((key) => {
-      args.push(`${key}: ${err[key]}`);
-    });
-    logHelper.error(err, `\n\n${args.join('\n')}`);
-    throw err;
-  })
-  // We must give the generated stream the same name as the entry file
-  // for the sourcemaps to work correctly
-  .pipe(source(moduleIndexPath))
-  // gulp-sourcemaps don't work with streams so we need
-  .pipe(buffer())
-  // This tells gulp-sourcemaps to load the inline sourcemap
-  .pipe(sourcemaps.init({loadMaps: true}))
-  // This renames the output file
-  .pipe(rename(outputFilename))
-  // This writes the sourcemap alongside the final build file
-  .pipe(sourcemaps.write('.'))
-  .pipe(gulp.dest(outputDirectory));
-};
-
-const cleanPackages = (packagePath) => {
+const cleanPackage = (packagePath) => {
   const outputDirectory = path.join(packagePath,
-    constants.PACKAGE_BUILD_DIRNAME, constants.BROWSER_BUILD_DIRNAME);
-  return fs.remove(outputDirectory);
+    constants.PACKAGE_BUILD_DIRNAME);
+  return fse.remove(outputDirectory);
 };
 
 gulp.task('build-packages:clean', gulp.series(
-  packageRunnner('build-packages:clean', cleanPackages)
+    packageRunnner.wrapFunction(
+      'Cleaning Package',
+      packageRunnner.getPackages(),
+      cleanPackage
+    )
 ));
 
-// This will create one version of the tests for each buildType.
-// i.e. we'll have a browser build for no NODE_ENV and one for 'production'
-// NODE_ENV and the same for sw and node tests.
-const packageBuilds = constants.BUILD_TYPES.map((buildType) => {
-  return packageRunnner('build-package', buildPackage, buildType);
-});
+gulp.task('build-packages:build', (done) => {
+  const buildStreams = [];
 
-gulp.task('build-packages:build', gulp.series(packageBuilds));
+  const nodePackages = packageRunnner.getPackages('node');
```

Comment:
It feels weird requiring us to get the packages and pass that around vs, vs `packageRunner.runOnBrowserPackages('<task>', buildNodePackage);`

Reply:
There are tasks which need to be run regardless of which `packageType` we're dealing with, like `build-packages:clean`.

Having a method that could potentially return everything, but also could be filtered down when needed, made the most sense to me.

With your approach, we'd either end up with

```js
packageRunner.runOnBrowserPackages('build-packages:clean', cleanPackage);`
packageRunner.runOnNodePackages('build-packages:clean', cleanPackage);`
```

or creating a new `packageRunner.runOnNodeAndBrowserPackages()` method, neither of which I find appealing.

Reply:
How about `packagerunner.run('<task name>', 'node'|'browser'|'all', func)`

Reply:
If you're happy with that, we can go with it.